File: gateOne/menstrualWahala/menstrual_app.py
from datetime import datetime, timedelta

date_format = "%d-%m-%Y"

previous_start_date = input("Enter the start date of your previous or current Menstrual period \"use the format (dd-MM-yyyy)\": ")
print()

last_start_date = datetime.strptime(previous_start_date, date_format).date()
formatted_lastStartDate = last_start_date.strftime("%d-%m-%Y")


menstrual_cycle = int(input("On the average, how long is your typical Menstrual Cycle: "))
print()



menstrual_duration = int(input("how many days did your period use to last: "))
print()



last_end_date = last_start_date + timedelta(days = menstrual_duration - 1)

next_start_date = last_start_date + timedelta(days = menstrual_cycle)

next_end_date = next_start_date + timedelta(days = menstrual_duration - 1)

print("The estimated date of your next period is ", next_start_date.strftime("%d-%m-%Y"), " to ", next_end_date.strftime("%d-%m-%Y"))
print()

ovulation_start_date = last_start_date + timedelta(days = menstrual_cycle / 2 - 1)

ovulation_end_date = ovulation_start_date + timedelta(days = 1)

print("Your estimated Ovulation Date is ", ovulation_start_date.strftime("%d-%m-%Y"), " to ", ovulation_end_date.strftime("%d-%m-%Y"));
print()

first_safe_period_start = last_end_date + timedelta(days = 1)


first_safe_period_end = first_safe_period_start + timedelta(days = 3)

second_safe_period_start = ovulation_end_date + timedelta(days = 5)

second_safe_period_end = second_safe_period_start + timedelta(days = 4)

print("Your estimated Safe Period is ", first_safe_period_start.strftime("%d-%m-%Y"), " to ", first_safe_period_end.strftime("%d-%m-%Y"), " and ", second_safe_period_start.strftime("%d-%m-%Y"), " to ", second_safe_period_end.strftime("%d-%m-%Y"));

File: gateOne/menstrualWahala/MenstrualApp.java
import java.time.LocalDate;
import java.util.Scanner;
import java.time.format.DateTimeFormatter;

public class MenstrualApp{
	public static void main(String[] args){
		Scanner input = new Scanner(System.in);

		LocalDate today = LocalDate.now();

		DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");



	
	
		System.out.print("Enter the start date of your previous or current Menstrual period \"use the format (dd-MM-yyyy)\": ");
		String previousStartDate = input.next();

		LocalDate lastStartDate = LocalDate.parse(previousStartDate, formatter);

		System.out.print("On the average, how long is your typical Menstrual Cycle: ");
		int menstrualCycle = input.nextInt();





		System.out.print("how many days did your period use to last: ");
		int menstrualDuration = input.nextInt();


		LocalDate lastEnddate = lastStartDate.plusDays(menstrualDuration - 1);
		System.out.println("The End date of your previous period was " + lastEnddate.format(formatter));

		System.out.println();
		LocalDate nextStartdate = lastStartDate.plusDays(menstrualCycle);

		LocalDate nextEndDate = nextStartdate.plusDays(menstrualDuration - 1);
		System.out.println("The estimated date of your next period is " + nextStartdate.format(formatter) + " to " + nextEndDate.format(formatter));

		System.out.println();
		LocalDate ovulationStartDate = lastStartDate.plusDays(menstrualCycle / 2 -1);

		LocalDate ovulationEnddate = ovulationStartDate.plusDays(1);
		System.out.println("Your estimated Ovulation Date is " + ovulationStartDate.format(formatter) + " to " + ovulationEnddate.format(formatter));

		System.out.println();

		LocalDate firstSafePeriodStart = lastEnddate.plusDays(1);

		LocalDate firstSafePeriodEnd = firstSafePeriodStart.plusDays(3);

		LocalDate secondSafePeriodStart = ovulationEnddate.plusDays(5);

		LocalDate secondSafePeriodEnd = secondSafePeriodStart.plusDays(4);




		System.out.println("Your estimated Safe Period is " + firstSafePeriodStart.format(formatter) + " to " + firstSafePeriodEnd.format(formatter) + " and " + secondSafePeriodStart.format(formatter) + " to " + secondSafePeriodEnd.format(formatter));

		System.out.println();

	}

}


